cmake/usage/usage.cmake

```diff
@@ -0,0 +1,91 @@
+# TODO: Set to make when make is used as a generator
+set(CMAKE_MAKE_PROGRAM ninja)
+get_filename_component(generator ${CMAKE_MAKE_PROGRAM} NAME)
+
+set(arch_list
+  arc
+  arm
+  nios2
+  riscv32
+  x86
+  xtensa
+  )
+
+set(board_dir $ENV{ZEPHYR_BASE}/boards)
+
+foreach(arch ${arch_list})
+  set(board_arch_dir ${board_dir}/${arch})
+
+  # Match the *_defconfig files in the board directories to make sure
+  # we are finding boards, e.g.  qemu_xtensa/qemu_xtensa_defconfig
+  file(GLOB_RECURSE defconfigs_for_${arch}
+    RELATIVE ${board_arch_dir}
+    ${board_arch_dir}/*_defconfig
```

Comment:
looks for <board>.yaml instead, not defconfig. It would be nice if we can actually parse the yaml file for the human readable board name and put it in the help instead of the identifier :-)

Reply:
It was written like this b/c KBuild was looking for _defconfig, I'll re-write it to look for yaml.

But I would prefer to push 'parsing yaml' to the backlog if you don't mind :)

Reply:
yes, backlog is fine as well, I can also take a look at that if chance permits

Reply:
@nashif : After changing to searching for ".yaml" instead of "_defconfig" i found that  arduino_101_mcuboot was dropped from the list of boards. Is that OK?

I see _mcuboot has a defconfig, but not a yaml file.

Reply:
yes, thats fine,.